### PR TITLE
web3connect/burnerwallet

### PR DESCRIPTION
**Pull request by dkent600**
Resolves: https://github.com/daostack/alchemy/issues/1342
Resolves: https://github.com/daostack/alchemy/issues/1320

Upgrade latest Web3Connect and add the BurnerWallet provider.

- new behavior: when the user changes the account in MM they are completely logged off in Alchemy and the cached account info is deleted (before we only reinitiatlized Arc)
- the latest Web3Connect replaces our provider caching code with their own

### CODE REVIEW

File: src/arc.ts [hunk 1]
```diff
@@ -281,6 +294,41 @@ async function ensureCorrectNetwork(provider: any): Promise<void> {
...
+     * the user can rescan a qrcode when changing WalletConnect provider.
+     */
+  if (selectedProvider && selectedProvider.close) {
+    selectedProvider.close(); // no need to await
+  }
+}
+
+export function getCachedAccount(): Address | null {
+  return localStorage.getItem(ACCOUNT_STORAGEKEY);
+}export interface IEnableWalletProviderParams {
```
Comment (tibetsprague) [inline src/arc.ts]: new line please

File: src/arc.ts [hunk 2]
```diff
@@ -295,23 +343,23 @@ function inTesting(): boolean {
 /**
  * Prompt user to select a web3Provider and enable their account.
  * Initializes Arc with the newly-selected web3Provider.
- * Is a no-op if already done.
+ * No-op if `selectedProvider` is already set (one can manuallygo to readonly mode to clear it)
```
Comment (tibetsprague) [inline src/arc.ts]: manually go

File: src/arc.ts (hunk 1 above)
Comment (dkent600) [reply src/arc.ts]: Done

File: src/arc.ts (hunk 2 above)
Comment (dkent600) [reply src/arc.ts]: done

**Pull request merged**